### PR TITLE
Added filled angle Example

### PR DESCRIPTION
**Pull request by kolibril13**
## Changelog / Overview
Useful example for advanced use of Angle

## Checklist
- [ ] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [ ] I have written a descriptive PR title (see top of PR template for examples)
- [ ] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/mobject/geometry.py [hunk 1]
```diff
@@ -2270,6 +2270,23 @@ def construct(self):
                 self.add(
                     line_list
                 )
+    .. manim:: FilledAngle
+        :save_last_frame:
+
+        class Fill(Scene):
```
Comment (PhotonSpheres) [inline manim/mobject/geometry.py:2276]: I think the class has to be called `FilledAngle` as in line 2273 for it to work.

Comment (kolibril13) [reply manim/mobject/geometry.py:2276]: yess :)

Comment (kolibril13) [inline manim/mobject/geometry.py:2276]:
```suggestion
        class FilledAngle(Scene):
```

**Comment by PhotonSpheres** [comment]
It might be cool to add the option of filling an angle to the class Angle in the first place. Like a `fill_angle=True` and `fill_color=...`, `fill_opacity=...` option. What do you think?

**Comment by kolibril13** [comment]
> It might be cool to add the option of filling an angle to the class Angle in the first place. Like a `fill_angle=True` and `fill_color=...`, `fill_opacity=...` option. What do you think?
Yess, absolutely!
And maybe also the parameter `fill_to_radius` so that it is possible to not fill the whole angle, but only a section.
Also there is already this class, which might already have all of these options:
https://docs.manim.community/en/stable/reference/manim.mobject.geometry.Annulus.html

**Comment by huguesdevimeux** [comment]
I _think_ it'd be cool to have some comments in the example for the important line. 

(if you agree, of course)

**Comment by PhotonSpheres** [comment]
> > It might be cool to add the option of filling an angle to the class Angle in the first place. Like a `fill_angle=True` and `fill_color=...`, `fill_opacity=...` option. What do you think?
> 
> Yess, absolutely!
> And maybe also the parameter `fill_to_radius` so that it is possible to not fill the whole angle, but only a section.
> Also there is already this class, which might already have all of these options:
> https://docs.manim.community/en/stable/reference/manim.mobject.geometry.Annulus.html
> EDIT:
> Annulus has no option to specify the angles, so I think that won't be the way to go:
> 
> ```python
> class DDo(Scene):
>     def construct(self):
>         self.add(Annulus(inner_radius=1, outer_radius=2)) # args (start_angle=0,angle=TAU / 4 ) not possible
>         self.wait()
> ```

I hacked something together in my `filled_angle` branch on my fork of manim. Try execute
 ```python
class AngleTest(Scene):
    def construct(self):
        l1 = Line(ORIGIN, RIGHT)
        l2 = Line(ORIGIN, UP)
        a = Angle(l1,l2, other_angle=True, fill_angle=True, fill_opacity=0.6, fill_color=GRAY, color=RED, stroke_width=5)
        self.add(l1,l2)
        self.play(DrawBorderThenFill(a))
```
and let me know what you think!

**Comment by behackl** [comment]
What is the status of this PR?

**Comment by jsonvillanueva** [comment]
> 
> 
> What is the status of this PR?

Bump! @kolibril13 
Do you intend to incorporate @PhotonSpheres suggestion?

**Comment by kolibril13** [comment]
Sorry for being inactive here, I was exchanging ideas with @PhotonSpheres in private messages about how we can add filled angles in a nicer way, my idea would be something like this:
```
​
class FilledAngle(VMobject):
    def __init__(self, a1, a2, **kwargs):
        VMobject.__init__(self, **kwargs)
​
        q1 = a1.get_points()
        q2 = a2.reverse_direction().get_points()
        pnts = np.concatenate([q1, q2, q1[0].reshape(1, 3)])
        self.set_color(D_GREEN)
        self.set_points_as_corners([q for q in pnts]).set_fill(GREEN, opacity=1)
        self.c1 = q1[0]
        self.c2 = q1[-1]
        self.c3 = q2[0]
        self.c4 = q2[-1]
        
    def get_corners(self):
        return (self.c1,self.c2,self.c3,self.c4)
```
and he had the idea of something like this in the `Angle` class:
```
if fill_angle:
                fill = AnnularSector(
                    arc_center=inter,
                    stroke_width=0,
                    stroke_opacity=0,
                    fill_opacity=fill_opacity,
                    color=fill_color,
                    start_angle=start_angle,
                    angle=angle_fin,
                    inner_radius=0,
                    outer_radius=radius)
                self.add(fill)
```
https://github.com/PhotonSpheres/manim/blob/ba1c3ba12c96fb1e4e7c73d6c1283531ddcc8535/manim/mobject/geometry.py#L2175

> Or perhaps @huguesdevimeux suggestion on comments? 

Yes, I will apply that.
In general, I think we can merge it soon and come back to it when there is a more profound way to make filled angles.

File: manim/mobject/geometry.py [hunk 2]
```diff
@@ -2270,6 +2270,23 @@ def construct(self):
...
+                norm=np.linalg.norm(l1.get_start() - l1.get_end())
+                a1=Angle(l1,l2, other_angle=True, radius= norm-0.5).set_color(GREEN)
+                a2=Angle(l1,l2, other_angle=True, radius= norm).set_color(GREEN)
+                q1=a1.get_points()
+                q2= a2.reverse_direction().get_points()
+                pnts= np.concatenate([q1,q2, q1[0].reshape(1,3)])
+                mfill = VMobject().set_color(ORANGE)
+                mfill.set_points_as_corners([q for q in pnts]).set_fill(GREEN, opacity=1)
+                self.add(l1,l2)
+                self.add(mfill)
```
Comment (kolibril13) [inline manim/mobject/geometry.py:2289]:
```suggestion
        class FilledAngle(Scene):
            def construct(self):
                l1 = Line(ORIGIN, 2 * UP + RIGHT).set_color(GREEN)
                l2 = (
                    Line(ORIGIN, 2 * UP + RIGHT)
                    .set_color(GREEN)
                    .rotate(-20 * DEGREES, about_point=ORIGIN)
                )
                norm = l1.get_length()
                a1 = Angle(l1, l2, other_angle=True, radius=norm - 0.5).set_color(GREEN)
                a2 = Angle(l1, l2, other_angle=True, radius=norm).set_color(GREEN)
                q1 = a1.get_points() #  save all coordinates of points of angle a1
                q2 = a2.reverse_direction().get_points()  #  save all coordinates of points of angle a1 (in reversed direction)
                pnts = np.concatenate([q1, q2, q1[0].reshape(1, 3)])  # adds points and ensures that path starts and ends at same point
                mfill = VMobject().set_color(ORANGE)
                mfill.set_points_as_corners(pnts).set_fill(GREEN, opacity=1)
                self.add(l1, l2)
                self.add(mfill)
```

**Review by jsonvillanueva — APPROVED** [review]
LGTM, although the example is a bit complex with ensuring that the point start/ends at the same coordinate. Definitely something to revisit!

**Pull request merged**